lib/snapshot-manager.js

```diff
@@ -405,30 +424,13 @@ function determineSnapshotDir({file, fixedLocation, projectDir}) {
 	}
 
 	return testDir;
-}
+}, {cacheKey: ([{file}]) => file});
 
-function resolveSourceFile(file) {
-	const testDir = path.dirname(file);
-	const buffer = tryRead(file);
-	if (!buffer) {
-		return file; // Assume the file is stubbed in our test suite.
-	}
-
-	const source = buffer.toString();
-	const converter = convertSourceMap.fromSource(source) || convertSourceMap.fromMapFileSource(source, testDir);
-	if (converter) {
-		const map = converter.toObject();
-		const firstSource = `${map.sourceRoot || ''}${map.sources[0]}`;
-		return path.resolve(testDir, firstSource);
-	}
-
-	return file;
-}
+exports.determineSnapshotDir = determineSnapshotDir;
 
 function load({file, fixedLocation, projectDir, recordNewSnapshots, updating}) {
-	const sourceFile = resolveSourceFile(file);
-	const dir = determineSnapshotDir({file: sourceFile, fixedLocation, projectDir});
-	const relFile = path.relative(projectDir, sourceFile);
+	const dir = determineSnapshotDir({file, fixedLocation, projectDir});
+	const relFile = path.relative(projectDir, resolveSourceFile(file));
```

Comment:
Note: In order to not have to export both `determineSnapshotDir()` and `resolveSourceFile()` I changed the signature of `determineSnapshotDir()` to take the "raw" file instead of the source file (it resolves the source file on its own).

Caveat: `resolveSourceFile()` is called twice for `load()` (once more to determine the relative path). But since it's memoized, I think it should be fine, no?

Reply:
I think that's fine.